### DISCUSSION TITLE
Prevent input from closing on submit

### DISCUSSION THREAD

**Discussion by danielo515** [MunifTanjim/nui.nvim#122] [Q&A]
Hello.
Is it possible to prevent an input from closing on submit?
I guess a possible workaround would be to update some internal state on change and bind a custom function to the CR key, but I was wondering if there is an easier way.

**Comment by MunifTanjim** [comment]
I think you're on the right track. You'll need to override the keybindings to acheive that. e.g.:

```lua
  input:map("i", "<CR>", function()
    -- do something
  end, { noremap = true })

  input:map("n", "<CR>", function()
    -- do something
  end, { noremap = true })
```

I'm curious about your use-case. What are you trying to implement? 🤔

**Reply by danielo515** [reply]
I'm doing a two sections "form". One text input and then a text area. When you submit the input it should jump to the text area, and then at the end it should return a table with the content of both (or call a callback with the data)

**Reply by MunifTanjim** [reply]
Got it. I think overriding keybindings is the way to go. To capture the value from the input, you can use the `on_change` callback: https://github.com/MunifTanjim/nui.nvim/tree/abdbfab89f307151db83b1a5147cd390ef27ff99/lua/nui/input#on_change

**Reply by danielo515** [reply]
That is what I was planning, but I wanted to know if there was any other way. Thank you
